### PR TITLE
SPOI-5053 APEX-56 #resolve #comment

### PR DESCRIPTION
**Pull request by ishark**
Fixing removal of terminated operators from physical plan when downanStream operators are also completed till shutdown window Id
Also fixed containers to be removed only when operators are removed from physical plan

### CODE REVIEW

File: engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java [hunk 1]
```diff
@@ -1040,15 +1040,31 @@ public int processEvents()
...
-            // wait until window is committed
-            continue;
-          } else {
+          if (windowAndOpers.getKey().longValue() <= this.committedWindowId) {
             LOG.info("Removing inactive operators at window {} {}", Codec.getStringWindowId(windowAndOpers.getKey()), windowAndOpers.getValue());
             for (PTOperator oper : windowAndOpers.getValue()) {
               plan.removeTerminatedPartition(oper);
             }
             it.remove();
+          } else {
```
Comment (tweise) [inline engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java]: Entry is never removed from map. Please do that and plan.remove in a single place.

File: engine/src/test/java/com/datatorrent/stram/StreamingContainerManagerTest.java [hunk 2]
```diff
@@ -762,18 +762,14 @@ public void testOperatorShutdown()
     mc1.sendHeartbeat();
     scm.monitorHeartbeat();
 
+    scm.monitorHeartbeat();
```
Comment (tweise) [inline engine/src/test/java/com/datatorrent/stram/StreamingContainerManagerTest.java]: Why is this added and with no preceding assertion?

File: engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java [hunk 3]
```diff
@@ -1743,11 +1757,10 @@ public void run()
 
     ContainerHeartbeatResponse rsp = getHeartbeatResponse(sca);
 
-    if (containerIdle && isApplicationIdle()) {
+    if (heartbeat.getContainerStats().operators.isEmpty() && isApplicationIdle() && containerIdle) {
```
Comment (tweise) [inline engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java]: remove containerIdle, it is no longer needed

**Comment by tweise** [comment]
Did you test shutdown-app from the CLI?

**Comment by ishark** [comment]
Yes, tested shutdown-app from CLI
